components/centraldashboard-angular/frontend/src/app/pages/main-page/main-page.component.ts

```diff
@@ -63,4 +83,45 @@ export class MainPageComponent implements OnInit, OnDestroy {
   computeBuildValue(label: string, buildValue: string): string {
     return `${label} ${buildValue}`;
   }
+
+  handleDashboardLinks(links: DashboardLinks) {
+    const { menuLinks, externalLinks, quickLinks, documentationItems } = links;
+    this.menuLinks = menuLinks || [];
+    this.externalLinks = externalLinks || [];
+    this.quickLinks = quickLinks || [];
+    this.documentationItems = documentationItems || [];
+  }
+
+  getUrlPath(url: string) {
+    const urlWithoutFragment = url.split('#')[0];
+    return this.appendPrefix(urlWithoutFragment);
+  }
+
+  getUrlFragment(url: string): string {
+    const fragment = url.split('#')[1];
+    return fragment;
+  }
+
+  appendPrefix(url: string): string {
+    return '/_' + url;
+  }
+
+  isLinkActive(url: string): boolean {
+    let browserUrl = this.router.url;
+    browserUrl = appendBackslash(browserUrl);
+    url = appendBackslash(url);
+    /**
+     * We need this condition because Pipelines Web App removes
+     * the 's' from its path when navigating inside a Recurring
+     * Run's details page
+     */
+    if (
+      browserUrl.includes('pipeline/#/recurringrun') &&
+      url.includes('/pipeline/#/recurringruns')
+    ) {
+      return true;
+    }
```

Comment:
It is ok for now, but we may need to do something better here. I would prefer not to have hardcoded values. Maybe we can have `pipeline/#/recurringrun` inside `menuLinks` and compare it with `startsWith()`.

Reply:
ACK I agree that this is not something we should tackle in the Dashboard's logic and having `if`s for different WAs isn't a good practice.
cc @kimwnasptd for awareness